app/bundles/CoreBundle/Model/GlobalSearchModalInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Model;
+
+use Doctrine\ORM\Tools\Pagination\Paginator;
+
+interface GlobalSearchModalInterface
+{
+    public function getEntitiesForGlobalSearch(string $searchString): ?Paginator;
```

Comment:
Do we need the paginator? There is no pagination, in the search, right? I think it performs additional count query to build the pagination features. Perhaps a collection could be used instead?

Reply:
Yes, we need the paginator to get the total number of results. We only show three results, and if more results are available, we show the "more" link, which navigates to the entity listing page.

Example, 
<img width="629" alt="Screenshot 2025-01-10 at 9 15 09 AM" src="https://github.com/user-attachments/assets/ad47bbe1-edb7-4052-9449-bb7dc94e7e3c" />

Reply:
It's a good fit for this use case then 👍